### PR TITLE
Blog search functionality

### PR DESCRIPTION
**Pull request by andipavllo**
@argelius when you have time, please check this.

### CODE REVIEW

**Comment by argelius** [comment]
Thanks @andipavllo ! I will look at it tomorrow

**Comment by argelius** [comment]
@andipavllo Sorry for the delay. I think it would be better if the mouse pointer changes when hovering the magnifying glass.

File: src/files/scripts/blog_search.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    searchBar.classList.add("blog-header-search-bar-hidden");
+
+    searchContainer.classList.remove("blog-header-search-container-visible");
+    searchContainer.classList.add("blog-header-search-container-hidden");
+
+    document.removeEventListener("click", checkOuterClick);
+    searchIcon.removeEventListener("click", performSearch);
+    searchIcon.addEventListener("click", showSearchBar);
+  }
+})();
```
Comment (argelius) [inline src/files/scripts/blog_search.js]: Missing newline

**Comment by argelius** [comment]
One more thing. Does checking for keyCode 13 work on mobile browsers? Maybe it's better to wrap in a form and listen for `submit` event.

File: src/files/scripts/blog_search.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  function checkOuterClick(e) {
+    if (e.target != searchBar) {
+      hideSearchBar();
+    }
+  }
+
+  function performSearch(e) {
+    if (searchBar.value != "" && (e.keyCode==13 || e.target == searchIcon)) {
```
Comment (argelius) [inline src/files/scripts/blog_search.js]: Please use strict comparison

File: src/files/scripts/blog_search.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+      document.addEventListener("click", checkOuterClick);
+      searchBar.addEventListener("keypress", performSearch);
+      searchIcon.removeEventListener("click", showSearchBar);
+      searchIcon.addEventListener("click", performSearch);
+    }, 100);
+  }
+
+  function checkOuterClick(e) {
+    if (e.target != searchBar) {
```
Comment (argelius) [inline src/files/scripts/blog_search.js]: Strict comparison

File: src/files/scripts/blog_search.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    setTimeout(function() {
+      searchBar.classList.remove("blog-header-search-bar-hidden");
+      searchBar.classList.add("blog-header-search-bar-visible");
+
+      document.addEventListener("click", checkOuterClick);
+      searchBar.addEventListener("keypress", performSearch);
+      searchIcon.removeEventListener("click", showSearchBar);
+      searchIcon.addEventListener("click", performSearch);
+    }, 100);
```
Comment (argelius) [inline src/files/scripts/blog_search.js]: What is this delay for?

**Comment by andipavllo** [comment]
@argelius thank you for the feedback! I'll fix it the next couple of days.

**Comment by argelius** [comment]
Thanks!

**Comment by andipavllo** [comment]
@argelius I fixed the code, just two things:

* The search functionality will not be visible on mobile browsers, so we should not be worried about `keyCode 13`
* I was using the setTimeout because the event was being captured by another eventListener, causing undesired behavior. Fixed it by stopping the event propagation and removing the timeout.

**Comment by argelius** [comment]
Thanks @andipavllo !

One more small thing. I think there should be a hover effect just like the other links on the toolbar.

**Comment by andipavllo** [comment]
@argelius added, merge it if it's ok :)

**Comment by argelius** [comment]
Thanks @andipavllo !

**Pull request merged**